src/jquery.uls.languagefilter.js

```diff
@@ -207,15 +207,35 @@
 				var autofillLabel,
 					results = [];
 
-				$.each( result.languagesearch, function ( code, name ) {
-					if ( this.options.languages[ code ] ) {
+				$.each( result.languagesearch, function ( apiCode, name ) {
+					var code,
+						redirect = $.uls.data.isRedirect( apiCode );
```

Comment:
I don't like the fact we're checking for redirect every single one of language search results and doing that unconditionally, but will not make that a blocker. I would just like slightly reorganized code.

Reply:
It is a very cheap call, and this structure allows to keep the logic simple. Please show how would you do it better :)

Reply:
Yes, I agree that call is cheap. Achieving same, simple logic, with slightly rearranged structure is more appealing to me. Here is one way of doing it:

```
var code, redirect;

if ( this.options.languages[ apiCode ] ) {
    code = apiCode;
} else {
    redirect = $.uls.data.isRedirect( apiCode );

    if ( !redirect || !this.options.languages[ redirect ] ) {
        return;
    }

    code = redirect;
}
```